rows_test.go

```diff
@@ -75,3 +77,285 @@ func TestRowsClose(t *testing.T) {
 		t.Fatalf("rows close of driver was not called")
 	}
 }
+
+type rowsNextInterceptor struct {
+	NullInterceptor
+
+	rowsNextCalled  bool
+	rowsNextLastCtx context.Context
+}
+
+func (r *rowsNextInterceptor) RowsNext(ctx context.Context, rows driver.Rows, dest []driver.Value) error {
+	r.rowsNextCalled = true
+	r.rowsNextLastCtx = ctx
+	return rows.Next(dest)
+}
+
+func TestRowsNext(t *testing.T) {
+	con := &fakeConn{}
+	rows := &fakeRows{vals: [][]driver.Value{{"hello", "world"}}, con: con}
+	stmt := fakeStmt{
+		rows: rows,
+	}
+	con.stmt = stmt
+	driverName := t.Name()
```

Comment:
optional:

If `go test` is called with a `-count` parameter with a value >1 the testcases will fail because a driver with the same name is registered multiple times.
The test name is not unique enough, adding a microsec timestamp or a uuid could fix it.

The issue is not introduced with this PR and exists also in other testcases.
Therefore I don't see it as strictly necessary to be changed for the new testcases in this PR.

Reply:
One approach I've used for this kind of thing before is to have an atomic somewhere that you increment when generating a driver name.

Reply:
that approach is much better then my suggestion :-)